sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -80,6 +80,30 @@ class Sampler
    */
   virtual std::string GetDescription() const noexcept = 0;
 };
+
+class AlwaysOnSampler final : public sdk::trace::Sampler
+{
+public:
+  /**
+   * Returned SamplingResult always has RECORD_AND_SAMPLED
+   *    as the Decision
+   */
+  inline SamplingResult ShouldSample(
+      const SpanContext *parent_context,
+      trace_api::TraceId trace_id,
+      nostd::string_view name,
+      trace_api::SpanKind span_kind,
+      const trace_api::KeyValueIterable &attributes) noexcept override {
```

Comment:
We try to be consistent in [not naming unused parameters](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f9-unused-parameters-should-be-unnamed). See [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/trace/noop.h#L38) for an example.